src/main/java/io/pinecone/PineconeClientConfig.java

```diff
@@ -100,10 +100,6 @@ public PineconeClientConfig withServerSideTimeoutSec(int serverSideTimeout) {
     void validate() {
         if (apiKey == null)
             throw new PineconeValidationException("Invalid Pinecone config: missing apiKey");
-        if (environment == null)
```

Comment:
Removed these conditions from validate() since environment and projectName will be empty when connection url is used for data plane operations.